main_view.py

```diff
@@ -305,43 +305,44 @@ def mem_next_state(self):
             self.active_state += 1
             self.show_state(self.active_state)
 
-    @pyqtSlot(list)
+    @pyqtSlot(object)
     def set_active_pids(self, data):
         self.view_checked_pids(data)
-        if self.active_pids_len > 0:
-            self.set_buttons(data=True, refc=False, highlight=False)
+        self.set_buttons(data=True if self.active_pids_len > 0 else False,
+                         refc=False,
+                         highlight=False)
 
-    @pyqtSlot(list)
+    @pyqtSlot(object)
     def set_collection_time(self, data):
-        self.iteration_time = data[0] if data else -1
-        self.total_time = data[1] if data else -1
+        self.iteration_time = data[0] if data is not None else -1
+        self.total_time = data[1] if data is not None else -1
```

Comment:
You changed returned value, but we still working with it if it is a list. In my point of view this variant is not much more better than previous one

Reply:
Fair point, i just thought it would be better to return pair of values (tuple) instead of list in DynamicsDialog `get_values()` function, and None as indicator that smth went wrong